catalyst_voices/packages/internal/catalyst_voices_models/lib/src/proposal_schema/dtos/proposal_schema_dto.dart

```diff
@@ -0,0 +1,298 @@
+import 'package:catalyst_voices_models/src/proposal_schema/dtos/proposal_definitions_dto.dart';
+import 'package:catalyst_voices_models/src/proposal_schema/proposal_definitions.dart';
+import 'package:catalyst_voices_models/src/proposal_schema/proposal_schema.dart';
+import 'package:catalyst_voices_shared/catalyst_voices_shared.dart';
+import 'package:equatable/equatable.dart';
+import 'package:json_annotation/json_annotation.dart';
+
+part 'proposal_schema_dto.g.dart';
+
+@JsonSerializable()
+class ProposalSchemaDTO extends Equatable {
+  @JsonKey(name: r'$schema')
+  final String schema;
+  final String title;
+  final String description;
+  final ProposalDefinitionsDTO definitions;
+  final String type;
+  final bool additionalProperties;
+  @JsonKey(toJson: _toJsonProperties)
+  final List<ProposalSchemaSegmentDTO> properties;
+  @JsonKey(name: 'x-order')
+  final List<String> order;
+
+  const ProposalSchemaDTO({
+    required this.schema,
+    required this.title,
+    required this.description,
+    required this.definitions,
+    this.type = 'object',
+    this.additionalProperties = false,
+    required this.properties,
+    required this.order,
+  });
+
+  factory ProposalSchemaDTO.fromJson(Map<String, dynamic> json) {
+    final segmentsMap = json['properties'] as Map<String, dynamic>;
+    json['properties'] = segmentsMap.convertMapToListWithIds();
+
+    return _$ProposalSchemaDTOFromJson(json);
+  }
+
+  Map<String, dynamic> toJson() => _$ProposalSchemaDTOToJson(this);
+
+  ProposalSchema toModel() {
+    return ProposalSchema(
+      schema: '',
+      title: '',
+      description: '',
+      definitions: definitions.toModel(),
+      type: DefinitionsObjectTypes.fromString(type),
+      additionalProperties: false,
+      properties: [],
+      order: [],
+    );
+  }
+
+  @override
+  List<Object?> get props => [
+        schema,
+        title,
+        description,
+        definitions,
+        type,
+        additionalProperties,
+        properties,
+        order,
+      ];
+
+  static Map<String, dynamic> _toJsonProperties(
+    List<ProposalSchemaSegmentDTO> properties,
+  ) {
+    final map = <String, dynamic>{};
+    for (final property in properties) {
+      map[property.id] = property.toJson();
+    }
+    return map;
+  }
+}
+
+@JsonSerializable()
+class ProposalSchemaSegmentDTO extends Equatable {
+  @JsonKey(name: r'$ref')
+  final String ref;
+  final String id;
+  final String title;
+  final String description;
+  @JsonKey(toJson: _toJsonProperties)
+  final List<ProposalSectionDto> properties;
+  final List<String> required;
+  @JsonKey(name: 'x-order')
+  final List<String> order;
+
+  const ProposalSchemaSegmentDTO({
+    required this.ref,
+    required this.id,
+    this.title = '',
+    this.description = '',
```

Comment:
is `title` / `description` optional?

Reply:
Yeah it it. Sometimes we get it sometimes we don't.